drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/GuidedDecisionTableGraphEditorPresenter.java

```diff
@@ -108,28 +109,41 @@
 
     protected ObservablePath.OnConcurrentUpdateEvent concurrentUpdateSessionInfo = null;
     protected Access access = new Access();
+    protected Integer originalGraphHash;
 
     private class LoadGraphLatch {
 
         private int dtGraphElementCount;
-        private Path dtToSelectPath;
+        private Command onAllDocumentGraphEntriesLoadedCommand;
+        private ParameterizedCommand<GuidedDecisionTableView.Presenter> onDocumentGraphEntryLoadedCommand;
 
         private LoadGraphLatch( final int dtGraphElementCount,
-                                final Path dtToSelectPath ) {
+                                final ParameterizedCommand<GuidedDecisionTableView.Presenter> onDocumentGraphEntryLoadedCommand ) {
+            this( dtGraphElementCount,
+                  onDocumentGraphEntryLoadedCommand,
+                  () -> {/*Do nothing*/} );
+        }
+
+        private LoadGraphLatch( final int dtGraphElementCount,
+                                final ParameterizedCommand<GuidedDecisionTableView.Presenter> onDocumentGraphEntryLoadedCommand,
+                                final Command onAllDocumentGraphEntriesLoadedCommand ) {
             this.dtGraphElementCount = dtGraphElementCount;
-            this.dtToSelectPath = dtToSelectPath;
+            this.onDocumentGraphEntryLoadedCommand = onDocumentGraphEntryLoadedCommand;
+            this.onAllDocumentGraphEntriesLoadedCommand = onAllDocumentGraphEntriesLoadedCommand;
         }
 
-        private void fireDecisionTableSelectedEvent( final GuidedDecisionTableView.Presenter dtPresenter ) {
```

Comment:
Hi @jomarko I changed the ```LoadGraphLatch``` class to invoke callbacks when a graph entry (i.e. Decision Table) was loaded and when all entries had been loaded. This allowed me to set the "original graph hash" after the graph was loaded. I have however changed this PR to share a private method that builds the command to select a graph entry (Decision Table).